### DISCUSSION TITLE
XDrip & Dexcom Clarity upload

### DISCUSSION THREAD

**Discussion by jpsollie** [NightscoutFoundation/xDrip#3450] [Q&A] [answered]
Hi, first of all: thank you for your work on those apps! I currently use BYODA, but the fact that these ideas exist makes that I can keep my powerful (yet a little older) lineageos phone is awesome! Your initiatives are priceless!
The hospital in Belgium (EU) which provides me the G6 sensors connected to the insulin pump + BYODA now wants me to switch to G7.
Unfortunately, I have the 'no internet connection' issue in BYODA (As they mention in the FAQ), so I'm looking for a better alternative, and I hope you people could help me.
Can xDrip report data to Dexcom Clarity? The hospital providing the sensors expects me to fix that.

*EDIT*:  searching a 2nd time brought me this answer: https://github.com/NightscoutFoundation/xDrip/discussions/3139#discussion-5750680
Is this the feature I'm looking for?
I can search for a solution keeping myself on the G6 + BYODA for now if needed, but I'd really need the live monitoring from the hospital to be up & running next month, whether it's G6 (current setup) or G7, don't care.

**Comment by Navid200** [comment]
>  I have the 'no internet connection' issue in BYODA 

What do you mean by that?

**Reply by jpsollie** [reply]
https://www.patreon.com/posts/troubleshooting-84715418 -> part 'trouble logging in', issue 4.
I can't use a US account, it takes it impossible for the hospital to follow if the patient's account is in another country

**Reply by Navid200** [reply]
xDrip cannot upload to Clarity.  This is not because we don't want to upload to Clarity.  Let's leave it at that.

Why do you need to upload to Clarity?

**Reply by Navid200** [reply]
Can the hospital use Tidepool or Nightscout for following you?

**Reply by jpsollie** [reply]
Their dexcom contract does not even allow suggesting BYODA, so ...

**Reply by Navid200** [reply]
So, they insist that you use the official app; you are willing to use the official app; you have evidence that the official app has shortcomings that cuts off for people who don't live in the States.
Did I miss anything?

What do they suggest you do?
It seems to me if your hospital is imposing restrictions, they should offer solutions for the known shortcomings.

**Reply by Navid200** [reply, marked as answer]
xDrip's policy, not mine, is to offer as much flexibility for diabetics so that we help them if we can.
The price to pay is that we have some parts in the code that is extremely complicated and is hard to maintain.

I am sorry I don't meant to complain.  I just want to understand what options we may have available to assist you.
These are the facts:


1- xDrip cannot upload to Clarity.
2- xDrip can collect from G7.
3- For those who must upload to Clarity, they will need to either use the official Dexcom app or BYODA.
4- xDrip can get readings from the official app using the companion app mode (https://navid200.github.io/xDrip/docs/Follow/CompanionApp.html).
5- The G6 BYODA can broadcast to xDrip.  But, the G7 BYODA as far as I know cannot.  
6- G7 supports a 24-hour backfill.  This means if you get a G7 Dexcom receiver, which I believe is cheaper than a G6 receiver, you only need to turn it on once a day to get all the readings into the receiver.  Then, once a month, you can connect the receiver to a computer using a USB cable and upload to Clarity.
In this case, you can use xDrip as a G7 collector: https://navid200.github.io/xDrip/docs/Dexcom/G7.html

**Reply by jpsollie** [reply]
They can give me another insuline pump which doesn't use dexcom. Unfortunately, my experience with this one is good, so I'd like to keep that one. And keep this phone as well

**Reply by Navid200** [reply]
Do you know if you can temporarily and safely disable the pump's connection to the transmitter if/when needed?
What I suggested with respect to use the G7 receiver requires to connect to G7 once a day using the receiver.  But, to do that, the pump needs to not be connecting to G7 for 5-10 minutes while receiver establishes connectivity to it for a few minutes.

**Reply by jpsollie** [reply]
I'll try to figure out something. Thank you for your support!

**Comment by jpsollie** [comment]
Discussion closed as question has been answered
